tests/utils/test_yaml.py

```diff
@@ -84,3 +86,29 @@ def test_read_yaml_file_bad_decode(tmpdir, caplog):
     with pytest.raises(ValueError):
         read_yaml_from_file_path(config_path, 'schemas/container.json')
     assert "unable to decode JSON schema, cannot validate" in caplog.text
+
+
+@pytest.mark.parametrize(('config', 'expected'), [
+    ("""\
+        platforms: []
+        operator_manifests:
+            manifests_dir: "tmp"
+            enable_digest_pinning: true
+            repo_replacements: []
+    """, "at top level: validating 'anyOf' has failed ([] is not of type 'object', 'null')"),
+    ("""\
+        compose:
+            packages: []
+            pulp_repos: true
+        mage_build_method: "imagebuilder"
+    """, "at top level: validating 'anyOf' has failed\
+    (Additional properties are not allowed ('mage_build_method' was unexpected))"),
+])
+def test_read_yaml_validation_error(config, expected, caplog):
+    expected = re.sub(r'\s+', ' ', expected)
+
+    try:
+        read_yaml(config, 'schemas/container.json')
```

Comment:
```python
with pytests.raises(....):
    ...
```
Otherwise we may not know when `read_yaml` accidentally stops raising exceptions